### PR TITLE
feat:[CI-14661] Add allure report image

### PR DESCRIPTION
**Pull request by Fernando-Dourado**

### CODE REVIEW

File: custom-images/allure/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# vim:set ft=dockerfile:
+
+# Do not edit individual Dockerfiles manually. Instead, please make changes to the Dockerfile.template, which will be used by the build script to generate Dockerfiles.
+
+FROM harnesscommunity/ci-base-image:latest
+
+LABEL maintainer="Harness Community Engineering <community-engineering@harness.io>"
+
+ENV JAVA_VERSION 21.0.4
+ENV URL https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.4%2B7/OpenJDK21U-jdk_x64_linux_hotspot_21.0.4_7.tar.gz
```
Comment (hemanthmantri) [inline custom-images/allure/Dockerfile:10]: `JDK_DOWNLOAD_URL`

File: custom-images/allure/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		fontconfig \
+	&& \
+	sudo rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* && \
+	# The dual version command is to support OpenJDK 8
+	java --version || java -version && \
+	javac --version || javac -version
+
+ENV ALLURE_VERSION=2.30.0 \
+	PATH=/opt/allure/bin:$PATH
+RUN dl_URL="https://repo.maven.apache.org/maven2/io/qameta/allure/allure-commandline/${ALLURE_VERSION}/allure-commandline-${ALLURE_VERSION}.zip" && \
```
Comment (hemanthmantri) [inline custom-images/allure/Dockerfile:34]: Set all Env variables in the beginning. `ALLURE_DOWNLOAD_URL`

File: custom-images/allure/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ENV URL https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.4%2B7/OpenJDK21U-jdk_x64_linux_hotspot_21.0.4_7.tar.gz
+ENV JAVA_HOME /usr/local/jdk-${JAVA_VERSION}
+
+RUN curl -sSL -o java.tar.gz "${URL}" && \
+	sudo mkdir /usr/local/jdk-${JAVA_VERSION} && \
+	sudo tar -xzf java.tar.gz --strip-components=1 -C /usr/local/jdk-${JAVA_VERSION} && \
+	rm java.tar.gz && \
+	if [[ "$JAVA_VERSION" == *"0.0"* ]]; then \
+		sudo ln -s /usr/local/jdk-${JAVA_VERSION} /usr/local/jdk-21; \
+	fi && \
```
Comment (hemanthmantri) [inline custom-images/allure/Dockerfile:19]: What is this conditional check for?

File: custom-images/allure/Dockerfile (hunk 2 above)
Comment (Fernando-Dourado) [reply custom-images/allure/Dockerfile]: Done

File: custom-images/allure/Dockerfile (hunk 1 above)
Comment (Fernando-Dourado) [reply custom-images/allure/Dockerfile]: Done

**Review by Ompragash — CHANGES_REQUESTED** [review]
- let’s also test this image. Please create a test image, push it, and run the Allure command to ensure it’s running as expected.
- update the README.md Development Status table with the Allure image entry.

**Comment by Fernando-Dourado** [comment]
Image build: https://app.harness.io/ng/account/gCoPSwHxS7ipOgx2iA9tOQ/module/ci/orgs/default/projects/CI_Security_Test/pipelines/testallureimage/executions/OqapPmV0Sfmd49bsitIETA/pipeline?step=jtUC_UW9QU2pfi8NyoB0kw&stage=2ovlN0BrTpmGLpsrYtSJKQ&childStage=&stageExecId=

Image validation: https://app.harness.io/ng/account/gCoPSwHxS7ipOgx2iA9tOQ/module/ci/orgs/default/projects/jenkinstoharness/pipelines/Allure/executions/vyS0a2lfSlqWfW2QwvyebA/pipeline

**Review by Ompragash — APPROVED** [review]

**Pull request merged**